zgw/portaaltaak/src/main/kotlin/com/ritense/portaaltaak/PortaalTaakEventListener.kt

```diff
@@ -0,0 +1,178 @@
+        documentService.modifyDocument(
+            ModifyDocumentRequest(
+                document.id().toString(),
+                jacksonObjectMapper().valueToTree(taakObject.verzondenData),
+                document.version().toString()
+            )
+        ).also { result ->
+            if (result.errors().size > 0) {
+                throw NotificatiesException(
+                    "Could not update document" +
+                            "Reason:\n" +
+                            result.errors().joinToString(separator = "\n - "),
+                    HttpStatus.INTERNAL_SERVER_ERROR
+                )
+            }
+        }
+    }
+
+    internal fun getDocumentenUrls(verzondenData: JsonNode): List<String> {
+        if (verzondenData.isMissingNode || verzondenData.isNull) {
+            return emptyList()
+        }
+        if (!verzondenData.isArray) {
+            throw RuntimeException("Not an array: '/documenten'")
+        }
+        val documentenUris = mutableListOf<String>()
+        for (documentPathNode in verzondenData) {
+            val documentUrlNode = verzondenData.at(JsonPointer.valueOf(documentPathNode.textValue()))
+            if (!documentUrlNode.isMissingNode && !documentUrlNode.isNull) {
+                try {
+                    if (documentUrlNode.isTextual) {
+                        documentenUris.add(documentUrlNode.textValue())
+                    } else if (documentUrlNode.isArray) {
+                        documentUrlNode.forEach { documentenUris.add(it.textValue()) }
+                    } else {
+                        throw RuntimeException("Invalid URL in '/documenten'. ${documentUrlNode.toPrettyString()}")
+                    }
+                } catch (e: MalformedURLException) {
+                    throw RuntimeException("Malformed URL in: '/documenten'", e)
+                }
+            }
+        }
+        return documentenUris
+    }
+
+    internal fun startProcessToUploadDocuments(
+        taakObject: TaakObject,
+        processDefinitionKey: String,
+        businessKey: String,
+        objectenApiPluginConfigurationId: String,
+        portaalTaakObjectResourceUrl: String
+    ) {
+        val variables = mapOf(
+            "portaalTaakObjectResourceUrl" to portaalTaakObjectResourceUrl,
+            "objectenApiPluginConfigurationId" to objectenApiPluginConfigurationId,
+            "verwerkerTaakId" to taakObject.verwerkerTaakId,
+            "documentUrls" to getDocumentenUrls(jacksonObjectMapper().valueToTree(taakObject.verzondenData))
+        )
+        try {
+            processService.startProcess(
+                processDefinitionKey,
+                businessKey,
+                variables
+            )
+        } catch (ex: RuntimeException) {
+            throw NotificatiesException(
+                "Could not start process with definition: $processDefinitionKey and businessKey: $businessKey.\n " +
```

Comment:
Should be a more specific exception

Reply:
here i'm just capturing the exception that startProcess might throw.
in the line bellow i'm appending the exception message which should provide more context on what actually went wrong